sasl/external/src/test/java/org/wildfly/security/sasl/external/ExternalSaslServerTest.java

```diff
@@ -180,12 +181,18 @@ public void testAuthnClientData() throws Exception {
     /**
      * Test failing (as we only authenticate "admin") authn for unsupported data "test" from client.
      */
-    @Test(expected = SaslException.class)
+    @Test
     public void testFailedAuthn() throws Exception {
-        SaslServer saslServer = obtainSaslServerFactory(ExternalSaslServerFactory.class).createSaslServer(EXTERNAL, "test",
-                "localhost", setProps(), CALLBACK_HANDLER_AUTHZ_ADMIN);
-        assertFalse(saslServer.isComplete());
-        saslServer.evaluateResponse("test".getBytes(StandardCharsets.UTF_8));
+        try {
+            SaslServer saslServer = obtainSaslServerFactory(ExternalSaslServerFactory.class).createSaslServer(EXTERNAL, "test",
+                    "localhost", setProps(), CALLBACK_HANDLER_AUTHZ_ADMIN);
+            Assert.assertThrows(SaslException.class,()->{
+                assertFalse(saslServer.isComplete());
+                saslServer.evaluateResponse("test".getBytes(StandardCharsets.UTF_8));
```

Comment:
@keshav-redhat Just a minor, if  `saslServer.evaluateResponse("test".getBytes(StandardCharsets.UTF_8));` throws an exception, the `assertFalse` is not needed since it won't get executed. This similar case as above in OAuth2TokenSecurityRealmTest.java where you removed `assertFalse`